common/peerresolver/peerResolver.go

```diff
@@ -100,3 +102,20 @@ func (pr PeerResolver) GetAllPeers() ([]string, error) {
 	}
 	return peers, nil
 }
+
+// FromRedirectKey resolve the peer from types.RedirectKey by a specific order
+func (pr PeerResolver) FromRedirectKey(key types.RedirectKey) (string, error) {
```

Comment:
I guess the other `PeerResolver.From*` functions on can be made private once all usage is refactored. Is that the plan?